src/test/java/org/jsoup/select/TraversorTest.java

```diff
@@ -0,0 +1,48 @@
+package org.jsoup.select;
+
+import static org.junit.Assert.assertEquals;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Node;
+import org.junit.Test;
+
+public class TraversorTest {
+  // Note: NodeTraversor.traverse(new NodeVisitor) is tested in ElementsTest#traverse() 
+  
+  @Test public void traverseFiltered() {
+    Document doc = Jsoup.parse("<body><div><p>Hello</p></div><div>There</div></body>");
+    final StringBuilder accum = new StringBuilder();
+    NodeTraversor.traverse(new NodePartialVisitor() {
+        public boolean head(Node node, int depth) {
+            accum.append("<" + node.nodeName() + ">");
+            if ("p".equals(node.nodeName()))
+              return false; // OMIT contents of p.
+            return true;
+        }
+
+        public void tail(Node node, int depth) {
+            accum.append("</" + node.nodeName() + ">");
+        }
+    }, doc.select("body").first());
+    assertEquals("<body><div><p></p></div><div><#text></#text></div></body>", accum.toString());
+  }
+
+  @Test public void traverseFiltering() {
+    Document doc = Jsoup.parse("<body><div><p>Hello</p></div><div>There be <b>bold</b></div></body>");
+    NodeTraversor.traverse(new NodeFilteringVisitor() {
+        public boolean head(Node node, int depth) {
+            if ("p".equals(node.nodeName()))
+              return false; // DELETE p in head
+            return true;
+        }
+
+        public boolean tail(Node node, int depth) {
+          if ("b".equals(node.nodeName()))
+            return false; // DELETE b in tail
+            return true;
```

Comment:
In this case, it could be reduced to `return !"b".equals(node.nodeName());` as we don't have further rules. This is stripped down, I usually have multiple conditions.